torch/csrc/distributed/rpc/init.cpp

```diff
@@ -547,13 +547,14 @@ PyObject* rpc_init(PyObject* _unused, PyObject* noargs) {
                        const c10::intrusive_ptr<::c10d::ProcessGroup>& pg,
                        int numSendRecvThreads,
                        std::chrono::milliseconds rpcTimeout) {
-        return std::make_unique<ProcessGroupAgent>(
+        return std::shared_ptr<ProcessGroupAgent>(new ProcessGroupAgent(
```

Comment:
Why not keep a unique_ptr here with a custom deleter?

Reply:
If you look at the class definition above, the class is actually a `shared_ptr`. So what they were previously doing here was creating a unique ptr, and then using the stdlib functionality that lets you steal into a shared pointer. So you could do a unique ptr with a custom deleter but it would be pointless, it's equivalent to what's written here (but what is written here is a lot more direct).